third_party/window/window.go

```diff
@@ -0,0 +1,87 @@
+// source: https://github.com/jbrukh/window
+package store
+
+// An array-based moving window; a moving window
+// is a queue with a maximum size and the property
+// that when the size is reached, pushing a new
+// element into the queue causes the head to be
+// popped.
+//
+// You can optimize the amount of slice copying
+// that the MovingWindow will be doing by trading
+// off with space complexity. Namely, the underlying
+// array is allocated with a size that is the
+// multiple of the intended capacity of the queue
+// so that copying is less frequent.
+type MovingWindow struct {
+	arr  []interface{}
```

Comment:
Storing a pointer to hourEntry as a float64 :)

On Mon, Aug 17, 2015 at 3:39 PM, Daniel Martí <notifications@github.com>
wrote:

> In third_party/window/window.go
> <https://github.com/kubernetes/heapster/pull/471#discussion_r37245593>:
>
> > +package store
> > +
> > +// An array-based moving window; a moving window
> > +// is a queue with a maximum size and the property
> > +// that when the size is reached, pushing a new
> > +// element into the queue causes the head to be
> > +// popped.
> > +//
> > +// You can optimize the amount of slice copying
> > +// that the MovingWindow will be doing by trading
> > +// off with space complexity. Namely, the underlying
> > +// array is allocated with a size that is the
> > +// multiple of the intended capacity of the queue
> > +// so that copying is less frequent.
> > +type MovingWindow struct {
> > +	arr  []interface{}
>
> @vishh <https://github.com/vishh> what do you mean? Storing float64
> instead of hourEntry?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/heapster/pull/471/files#r37245593>.
>